ui/app/search/search-ctrl.js

```diff
@@ -26,30 +26,22 @@
         var oldParams = mlSearch.getParams();
         var newParams = mlSearch.getCurrentParams();
 
-        // restore params, if we have cached results, and no params provided
-        if (model.search.results && _.isEqual( {}, newParams )) {
+        // capture URL params in mlSearch and ctrl model
+        mlSearch.fromParams().then(function() {
 
-          $location.search( oldParams );
-
-        } else {
-
-          // capture URL params in mlSearch and ctrl model
-          mlSearch.fromParams().then(function() {
-
-            // and run search if no cached results, or params changed
-            if (!model.search.results || !_.isEqual( oldParams, newParams )) {
-              search();
-            }
-
-          });
+          // and run search if no cached results, or params changed
+          if (!model.search.results || !_.isEqual( oldParams, newParams )) {
+            search(model.mlSearch.getCurrentParams().q);
+          }
 
-        }
+        });
 
         // capture URL params (forward/back, etc.)
         $scope.$on('$locationChangeSuccess', function(e, newUrl, oldUrl){
           if (newUrl !== oldUrl) {
             mlSearch.locationChange( newUrl, oldUrl ).then(function() {
-              search();
+              //TODO: how do we reflect it back to the directive?
+              search(model.mlSearch.getCurrentParams().q);
```

Comment:
The model is no longer in synch with the url. If i am not mistaken, the alternative is to inject remoteInput, SearchModel, etc and synch them all beforehand.